### PR TITLE
add calc_t.py

### PR DESCRIPTION
**Pull request by pigriver123**
I have add a function which is used to compute the t values of beta_gain and beta_loss.

### CODE REVIEW

**Comment by BenjaminHsieh** [comment]
Do you need the following imports? If not, you can delete them from the function file

import matplotlib.pyplot as plt 
import nibabel as nib
import os
from stimuli import events2neural
from scipy.stats import gamma
import math
from scipy.stats import mstats
from scipy.stats import t

**Comment by pigriver123** [comment]
Yeah,you are right.i will correct that

File: code/utils/calc_t.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    y_hat = a.dot(c)
+    residuals = b - y_hat
+    ma=npl.pinv(a.transpose().dot(a))
+    RSS = np.sum(residuals ** 2)
+    df = a.shape[0] - npl.matrix_rank(a)
+    MRSS = RSS / df
+    s2 = MRSS
+    v_cov = s2 * npl.inv(a.T.dot(a))
+    numerator1 = c[0]
+    denominator1 = np.sqrt(v_cov[0, 0])
```
Comment (BenjaminHsieh) [inline code/utils/calc_t.py:17]: I think you need to import numpy as np, you used it here,

can delete your 'scipy.stats' imports, right?

**Comment by BenjaminHsieh** [comment]
Also would be good if you can write some comments about the function, see other function scripts for example/format

File: code/utils/calc_t.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+
+import numpy.linalg as npl
+from scipy.stats import mstats
+from scipy.stats import t
+
+def significant(a,b,c):
```
Comment (briankleinqiu) [inline code/utils/calc_t.py:6]: Just to be clear it's for one voxel
 a =  entire design matrix X, b = y [:, i], c = beta[:,i] 
And then loop through for each voxel

**Comment by changsiyao** [comment]
This function is definitely good, just might want to add more comments, like what input, output would be. always good to keep good documentation.
Also, like Ben said, you need to import np, but can delete scipy.stats imports.

File: code/utils/calc_t.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    df=len(b)-2
```
Comment (BenjaminHsieh) [inline code/utils/calc_t.py:7]: can you comment on why this is len(b) -2 ? I know n-rank(x) is the degrees of freedom so why is x only rank 2? Maybe adding some comments can help as i am writing tests rn

Edit: nvm saw the other df line below, can you delete this line then?

**Comment by BenjaminHsieh** [comment]
Let us know if you need any help

**Comment by BenjaminHsieh** [comment]
Closing this for now because it has not been updated in a while, and so I can do some organization of functions

**Pull request closed without merging**